apps/src/blockly/addons/contextMenuAndShortcut.js

```diff
@@ -1,65 +1,16 @@
 import GoogleBlockly from 'blockly/core';
+import {CrossTabCopyPaste} from '@blockly/plugin-cross-tab-copy-paste';
 
-/**
- * Adds a copy command to the block context menu. After switching to v7,
- * we can replace this with:
- * https://github.com/google/blockly-samples/tree/master/plugins/cross-tab-copy-paste
- */
-const registerBlockCopyToStorage = function() {
-  const copyToStorageOption = {
-    displayText: function() {
-      return 'Copy';
-    },
-    preconditionFn: function(scope) {
-      if (scope.block.isDeletable() && scope.block.isMovable()) {
```

Comment:
This precondition is important because we want to prevent users from copying blocks they immovable/undeletable.

This is the behavior on levelbuilder today that we want to preserve:

https://user-images.githubusercontent.com/43474485/200342753-508738a7-25b5-4952-85b7-20740dd1aed4.mp4

To accomplish this, I think we'll need to override the `preconditionFn` function here: https://github.com/google/blockly-samples/blob/304a875648bac4c7d78747a298dc8b2a1d28ba5b/plugins/cross-tab-copy-paste/src/index.js#L67-L70

Reply:
I compared the  copying behavior with what's currently on Levelbuilder and how it's implemented in the plugin:
As you described, currently users are unable to copy and paste unmovable or undeletable blocks. Here's an example with an undeletable block:

https://user-images.githubusercontent.com/107423305/200419197-32275173-2e1e-47ae-b46e-3b5ef1df6b2a.mp4

With the plugin, users are able to copy and paste an unmovable or undeletable block, but when it's pasted, the settings for the new block are that it's movable and deletable. Here's an example with an undeletable block on this branch:

https://user-images.githubusercontent.com/107423305/200419461-c13a621d-9035-4ea8-9dce-29ebdd67c07f.mp4

I wonder if the behavior using the plugin implementation is actually more desirable for a user.

Reply:
After consulting with @maribethb, we agreed to follow the core Blockly behavior and not allow copying of immovable and undeletable blocks. @mikeharv 's [PR](https://github.com/google/blockly-samples/pull/1371) will fix the current inconsistency with the 'Copy' action.